### DISCUSSION TITLE
Footers displayed even when not in column definition with column group

### DISCUSSION THREAD

**Discussion by bastiengrignon** [TanStack/table#5023] [Q&A] [answered]
When using column grouping to have one header that take fullWidth of my table and all other columns below, but adding this header also add a footer below the table, thise footer is rendered but with empty element as you can see in the following [example](https://codesandbox.io/p/sandbox/festive-jones-3x7xzw?file=%2Fsrc%2Fmain.tsx%3A1%2C1)

I'd like to know if there is a way to not display theses footers ?

**Comment by bastiengrignon** [comment]
This seems to be an issue as this is also present in an example of tanstack v8 : https://tanstack.com/table/v8/docs/examples/react/column-groups

**Reply by rajagopalstalwart** [reply]
@bastiengrignon  this code is working, check here
<tfoot
                      style={{
                        position: "sticky",
                        bottom: 0,
                        margin: 0,
                        backgroundColor: "#f3f3f3",
                      }}
                    >
                      {table.getFooterGroups().map((footerGroup, index) => (
                        <tr key={footerGroup.id}>
                         {console.log("index111", index)}
                      {index === 0 && footerGroup.headers.map((header) => (
                            <th
                              key={header.id}
                              colSpan={header.colSpan}
                              style={{
                                border:"1px solid #616161",
                              }}
                            >
                              {header.isPlaceholder
                                ? null
                                : flexRender(
                                    header.column.columnDef.footer,
                                    header.getContext()
                                  )}
                            </th>
                          ))}
                        </tr>
                      ))}
                    </tfoot>

**Comment by xicrow** [comment]
This is still relevant, and kinda annoying :P

You don't even need to use grouping, just take the basic example from the documentation, and remove the footers:
https://tanstack.com/table/latest/docs/framework/react/examples/basic

Will still render a row for the footer with empty cells :/

**Reply by BrendanC23** [reply, marked as answer]
I'm running into the same issue. Here's a fix, though it isn't very elegant. This determines whether any of the columns actually have a footer. If not, you can return early from the component.

```ts
const footers = table
    .getFooterGroups()
    .map((group) => group.headers.map((header) => header.column.columnDef.footer))
    .flat()
    .filter(Boolean);

if (footers.length === 0) return;

// Render footer as normal
```

**Reply by farhan-hafizh** [reply]
hello, can you really provide me how exactly do you render the footer? @BrendanC23

**Reply by BrendanC23** [reply]
Take a look at [this example](https://tanstack.com/table/v8/docs/framework/react/examples/basic) from the TanStack Table docs. If none of the columns have a footer, you can return early to avoid rendering an empty footer.

```tsx
function Footer(table) {
    const footers = table
        .getFooterGroups()
        .map((group) => group.headers.map((header) => header.column.columnDef.footer))
        .flat()
        .filter(Boolean);

    if (footers.length === 0) return;

    return (
        <tfoot>
            {table.getFooterGroups().map((footerGroup) => (
                <tr key={footerGroup.id}>
                    {footerGroup.headers.map((header) => (
                        <th key={header.id}>
                            {header.isPlaceholder
                                ? null
                                : flexRender(header.column.columnDef.footer, header.getContext())}
                        </th>
                    ))}
                </tr>
            ))}
        </tfoot>
    );
}
```
